testplan/runners/pools/connection.py

```diff
@@ -2,46 +2,348 @@
 
 import pickle
 import warnings
-
+import six
+import abc
 import zmq
+import time
+from six.moves import queue
 
-from .base import ConnectionManager
+from testplan.common import entity
+from testplan.common.utils import logger
 
 
-class TCPConnectionManager(ConnectionManager):
+@six.add_metaclass(abc.ABCMeta)
+class Client(logger.Loggable):
     """
-    Manages pool-worker TCP communication.
+    Workers are Client in Pool/Worker communication.
+    Abstract base class for workers to communicate with its pool."""
+
+    def __init__(self):
+        super(Client, self).__init__()
+        self.active = False
+
+    @abc.abstractmethod
+    def connect(self, server):
+        """Connect client to server"""
+        self.active = True
+
+    @abc.abstractmethod
+    def disconnect(self):
+        """Disconnect client from server"""
+        self.active = False
+
+    @abc.abstractmethod
+    def send(self, message):
+        """
+        Sends a message to server
+        :param message: Message to be sent.
+        :type message: :py:class:`~testplan.runners.pools.communication.Message`
+        """
+        pass
+
+    @abc.abstractmethod
+    def receive(self):
+        """Receives response to the message sent"""
+        pass
+
+    def send_and_receive(self, message, expect=None):
+        """
+        Send and receive shortcut. Optionally assert that the response is
+        of the type expected. I.e For a TaskSending message, an Ack is expected.
+
+        :param message: Message sent.
+        :type message: :py:class:`~testplan.runners.pools.communication.Message`
+        :param expect: Assert message received command is the expected.
+        :type expect: ``NoneType`` or
```

Comment:
I'm a bit confused about the "expect" parameter - it is documented as being a `Message` type but is actually compared with the message's `cmd` string. Assuming it is meant to be the command, can we change the name to `expected_cmd` to be more explicit and update the documented type to `str`?